### DISCUSSION TITLE
Display the distance to the next speed limit

### DISCUSSION THREAD

**Discussion by langerheiko** [osmandapp/OsmAnd#11839] [Ideas]
I try to drive as energy-efficient as possible.
Therefore it would be nice to know when to take your foot off the accelerator. 
In addition to the distance, you should also be able to see the new speed limit before it applies.

Examples (from Germany):
- You are on the highway and have to brake from unlimited speed to 120 km/h.
- In the city you are allowed to drive 50km/h generally, but near schools or kindergartens speed limit is 30km/h
- You can drive overland at 100 km / h, but only 50 km / h in the next town - when should I brake?

**Comment by vshcherb** [comment]
That doesn't sound right cause OsmAnd has no intention to duplicate all Road Signs / Information for the driver, could be too dangerous. Indeed it's possible to see other road signs like Stop / Traffic Calming / ... but they are mostly done for OSM mappers and to represent upcoming road situation. 

It looks like this should be part of driving experience and infrastructure to indicate that soon will be another speed limit.

**Reply by x-500o** [reply]
Hi. I have similar needs.
When I approach of speedcamera I would also know to what limit of speed it is looking for. I can see cameras but it misses information on speed limits.
How can I enter such details beforehand so that next time it informs me of speed limit registered?

**Reply by mariush444** [reply]
the info about speed limit is shown in the widget. If exist in OSM of course.

**Reply by x-500o** [reply]
Thank you for your response.
Speed limit depends on a kind of road, but the installed speedcam may vary from that.
Am newer to OSM, how could I check what OSM contains on a particular road and its speadcams?
Would I be able to enrich any data or update its details for me?

**Reply by mariush444** [reply]
I'm regular user too
This is example that I found
https://www.openstreetmap.org/way/286066200#map=19/52.20415/20.25773&layers=D

it is something like "segmental speed measurement"
As I understand "maxspeed" tag should be mentioned.

But I think that someone with much more experience can provide more precise answer.
![image](https://user-images.githubusercontent.com/66887280/123337925-3ba59100-d548-11eb-9db9-d8f0d19016d2.png)

**Reply by mariush444** [reply]
@vshcherb 
Generally I agree with you about "duplicate all Road Signs". I always switch off notification about road cross or bumps.
But ...  Info about speed limit if the tag exists at speed camera seems to be reasonable. 
There is already big icon of speed camera then overlaying this icon with 2 or 3 digits could be nice feature.
my vote +1
